extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -311,30 +322,33 @@ public function getArticle() {
 		global $wgEnableMainPageDataMercuryApi;
 
 		try {
-			$title = $this->getTitleFromRequest();
-			$articleId = $title->getArticleId();
+			$title = $this->getRedirectTitle();
+			$data[ 'redirected' ] = true;
 
-			$data = [
-				'details' => $this->getArticleDetails( $articleId ),
-				'topContributors' => $this->getTopContributorsDetails(
-					$this->getTopContributorsPerArticle( $articleId )
-				)
-			];
+			if ( $title === null ) {
+				$title = $this->getTitleFromRequest();
+				unset ($data[ 'redirected' ] );
+			}
 
+			$articleId = $title->getArticleId();
```

Comment:
you are calling article->newFromID twice. Once inside getRedirectTitle, second time inside getArticleDetails.
Optimize it a little bit.